File: gamerpilotPlatform/Model/Course.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using gamerpilotPlatform.Model.Lectures;


namespace gamerpilotPlatform.Model
{
    public class Course
    {
        public string Id { get; set; }

        public string Name { get; set; }

        public string UrlName { get; set; }

        public string Description { get; set; }

        public string ImageUrl { get; set; }

        public Boolean IsReleased { get; set; }

        public virtual ICollection<Lecture> Lectures { get; set; }
        public virtual ICollection<CourseUser> EnrolledUsers { get; set; }

        [NotMapped]
        public virtual IEnumerable<Section> Sections { get; set; }


        public Course()
        {
            Lectures = new List<Lecture>();
            EnrolledUsers = new List<CourseUser>();
            Sections = new List<Section>();
        }
    }
}

File: gamerpilotPlatform/ClientApp/src/models/course.ts
import { Instructor } from './instructor';
import { Lecture } from './lecture';
import { Section } from './section';

export interface Course {
    id: string;
    description: string;
    imageUrl: string;
    isReleased: boolean;
    name: string;
    urlName: string;
    instructors: Instructor[];
    lectures: any[];
    sections: Section[];
    completedLectures: number[];
}
